src/_pytest/config/__init__.py

```diff
@@ -1178,8 +1178,13 @@ def notify_exception(
     def cwd_relative_nodeid(self, nodeid: str) -> str:
         # nodeid's are relative to the rootpath, compute relative to cwd.
         if self.invocation_params.dir != self.rootpath:
-            fullpath = self.rootpath / nodeid
-            nodeid = bestrelpath(self.invocation_params.dir, fullpath)
+            base_path_part = nodeid.split("::")[0]
+            nodeid_part = "::".join(nodeid.split("::")[1:])
+
+            fullpath = self.rootpath / base_path_part
+            relative_path = bestrelpath(self.invocation_params.dir, fullpath)
+
+            nodeid = f"{relative_path}::{nodeid_part}"
```

Comment:
Hi, @nauman897 This PR looks good !
A small suggestion about the coding stylem, which can avoid repeated use of the split method and accessing values ​​through indexes (of course, this is trivial)
```suggestion
            base_path_part, *nodeid_part = nodeid.split("::")
            # Only process path part
            fullpath = self.rootpath / base_path_part
            relative_path = bestrelpath(self.invocation_params.dir, fullpath)

            nodeid = "::".join([relative_path, *nodeid_part])
```

Reply:
Hey @dongfangtianyu ,
Thanks for the suggestion. I agree this looks better